pkg/gitproviders/provider.go

```diff
@@ -436,6 +438,47 @@ func (p defaultGitProvider) CreatePullRequestToOrgRepo(orgRepRef gitprovider.Org
 	return pr, nil
 }
 
+func (p defaultGitProvider) GetCommitsFromUserRepo(userRepRef gitprovider.UserRepositoryRef, targetBranch string) ([]gitprovider.Commit, error) {
+	ctx := context.Background()
+
+	ur, err := p.provider.UserRepositories().Get(ctx, userRepRef)
+	if err != nil {
+		return nil, fmt.Errorf("error getting info for repo [%s] err [%s]", userRepRef.String(), err)
+	}
+
+	// currently locking the commit list at 10. May discuss pagination options later.
+	commits, err := ur.Commits().ListPage(ctx, targetBranch, 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("error getting commits for repo[%s] err [%s]", userRepRef.String(), err)
+	}
+
+	if len(commits) == 0 {
```

Comment:
Is there ever a case where a branch has no commits? Is this truly an error?

Reply:
Yeah that was left from a test when i was having issues. Nice catch